capa/features/extractors/drakvuf/file.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Dict, List, Tuple, Iterator
+
+from capa.features.file import Import
+from capa.features.common import String, Feature
+from capa.features.address import NO_ADDRESS, Address, ThreadAddress, ProcessAddress, AbsoluteVirtualAddress
+from capa.features.extractors.helpers import generate_symbols
+from capa.features.extractors.base_extractor import ProcessHandle
+from capa.features.extractors.drakvuf.models import Call, DrakvufReport
+
+logger = logging.getLogger(__name__)
+
+
+def get_processes(calls: Dict[ProcessAddress, Dict[ThreadAddress, List[Call]]]) -> Iterator[ProcessHandle]:
+    """
+    Get all the created processes for a sample.
+    """
+    for proc_addr, calls_per_thread in calls.items():
+        sample_call = next(iter(calls_per_thread.values()))[0]  # get process name
```

Comment:
what's the benefit of `next(iter(...))` instead of `calls_per_thread.values()[0]`?

Reply:
I think I had this here cause I thought it'd be more performant (grabbing the first element from a generator). If I try to do what you suggested it does give me an error however:

![image](https://github.com/mandiant/capa/assets/16624109/8855deba-c5b0-4961-baee-937fe39563e9)

I think my reasoning was that doing `next(iter(...))` would better than doing `list(...)[0]`

Reply:
ah yeah, `.values()` gives a `dict_values` object so this makes sense